websearch/serper/src/lib.rs

```diff
@@ -0,0 +1,156 @@
+mod client;
+mod conversions;
+
+use std::cell::RefCell;
+
+use crate::client::{SearchRequest, SerperSearchApi};
+use crate::conversions::{params_to_request, response_to_results, validate_search_params};
+use golem_web_search::golem::web_search::web_search::{
+    Guest, GuestSearchSession, SearchError, SearchMetadata, SearchParams, SearchResult,
+    SearchSession,
+};
+
+use golem_web_search::LOGGING_STATE;
+
+struct SerperSearch {
+    client: SerperSearchApi,
+    request: SearchRequest,
+    params: SearchParams,
+    finished: bool,
+    metadata: Option<SearchMetadata>,
+    current_page: u32,
+}
+
+impl SerperSearch {
+    fn new(client: SerperSearchApi, request: SearchRequest, params: SearchParams) -> Self {
+        Self {
+            client,
+            request,
+            params,
+            finished: false,
+            metadata: None,
+            current_page: 1,
+        }
+    }
+
+    fn next_page(&mut self) -> Result<Vec<SearchResult>, SearchError> {
+        if self.finished {
+            return Ok(vec![]);
+        }
+
+        // Update request with current page
+        let request = self.request.clone();
+        // Note: Serper's SearchRequest doesn't have pagination fields
```

Comment:
Playground https://serper.dev/playground suggests that they have a 'page' parameter in the searchrequest